### DISCUSSION TITLE
直连连不上XTLS，已经测了端口没有被封的，请问是什么原理

### DISCUSSION THREAD

**Discussion by ghost** [XTLS/Xray-core#634] [Q&A] [answered]
使用443端口+websocket + tls + cdn, 可以正常连上；然后我想再在另一个端口（比如44444）开一个xtls直连的（即Vless + XTLS + TCP），直接使用ip连接，因为域名的话会被解析到cloudflare，是不可以直接xtls的。但是我直连XTLS却连不上。

这是因为cdn的缘故吗？为什么呢？我明明直接指定了ip啊，而且两边的allow insecure 都打开了

还是说最近墙太高，导致我没法向该vps进行tls握手？（我已经通过nc探测确认该端口没被封禁）

另外，SNI不管是填写域名，还是填写ip，还是根本不填写，都无法正常连接。

我在另一个vps的相同配置下，未配置cdn，地址直接填ip，然后SNI什么也不填，可以直接连上

**Comment by Xia-Jingye** [comment]
首先，跟cdn没有任何关系，其次，为什么不把xtls也配置在443端口呢？最后，我不知道你使用何种方式把xtls监听在44444端口上的，还请自行检查xray是否正常监听了44444端口。

**Comment by ghost** [comment]
> 
> 
> 首先，跟cdn没有任何关系，其次，为什么不把xtls也配置在443端口呢？最后，我不知道你使用何种方式把xtls监听在44444端口上的，还请自行检查xray是否正常监听了44444端口。

我同时监听了443，以及444xx-455xx的端口范围，端口范围的那个是单独的xtls，443就是各种回落那个。

我已经说了，我用nc -zv ip 44444 已经探测了，我的端口是正常打开的，因为探测未监听的端口会不通，探测监听的端口发现是通的，很明显已经正常监听了

**Comment by ghost** [comment]
用nc测试，进行传输数据，服务端能够接收到，用的相同的端口。但是xray连接的话，服务端的日志根本没有显示，就像没有人连一样。

我之前某端口被封了，nc测试该端口发现nc是正常的。估计是墙封端口并不是直接封，而是封该端口的协议。如果如此，可能我的vps的所有端口范围的xtls协议都被封了。似乎墙能精准识别xtls？？

**Comment by Butterflyflower** [comment]
CF的CDN之支持部分端口

**Reply by ghost** [reply]
我没用443，我用的是44444，没用cdn，就是说地址里没填域名。我的cdn的443还在用，好使，我想问的是不用cdn的44444端口

好好读一下我的情况把

**Comment by galaxyskyknight** [comment]
检查你vps的ufw或iptables

**Reply by ghost** [reply]
我防火墙是禁用的。我都说过了，“”端口被封了“”，意思就是“”端口之前都能用，后来不能用“”，你可以推理出没有防火墙问题

**Reply by ghost** [reply]
而且我都说了，用nc是能用该端口传输数据的，且是从国内的电脑向该节点传输数据，说明端口完全是好使的；唯一问题就是xray连不上

**Reply by ghost** [reply]
好好读一下我的情况把

**Reply by galaxyskyknight** [reply]
很简单，tcpdump

**Comment by ghost** [comment]
我的情况这2个人（HiColinn,horizon98）也存在，就是说端口没有“真正”被封，但是连不上xray
https://github.com/v2fly/v2ray-core/discussions/1113#discussioncomment-957391

**Comment by ghost** [comment]
现在443端口已经解封了，我可以更好地进行测试了，发现相同的配置的情况下，443端口不使用cdn是直接可以连上的，但是任何其他端口相同的配置就完全连不上。然后将日志打开，可以发现显示invalid request version, 也说明了这个端口，确实是能连上的，但是x RAY不知道为什么出现了问题

 这是什么情况？配置完全相同啊，我就是把443的部分的配置复制了一遍，然后把端口改一下，然后结果443好使，另外一个端口就不好使

还是说现在只解封了443端口，其他端口没有解封？

**Comment by ghost** [comment, marked as answer]
好蛋疼，重新复制一遍配置，只修改端口号，竟然又好使了。可能是编辑配置文件时使用了dos格式的问题吧。总之不是xray的问题
